src/main/cpp/chassis/driveStates/DriveToNote.cpp

```diff
@@ -0,0 +1,122 @@
+//====================================================================================================================================================
+// Copyright 2024 Lake Orion Robotics FIRST Team 302
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+// OR OTHER DEALINGS IN THE SOFTWARE.
+//=====================================================================================================================================================
+
+// C++ Includes
+
+// FRC Includes
+#include <frc/geometry/Rotation3d.h>
+#include <frc/geometry/Rotation2d.h>
+#include <pathplanner/lib/path/PathPlannerPath.h>
+
+// Team302 Includes
+#include "DragonVision/DragonVision.h"
+#include "chassis/SwerveChassis.h"
+#include "chassis/ChassisConfigMgr.h"
+#include "chassis/driveStates/DriveToNote.h"
+#include "utils/FMSData.h"
+#include "DragonVision/DragonVisionStructs.h"
+
+using namespace pathplanner;
+
+DriveToNote::DriveToNote() : m_chassis(nullptr)
+{
+    auto config = ChassisConfigMgr::GetInstance()->GetCurrentConfig();
+    m_chassis = config != nullptr ? config->GetSwerveChassis() : nullptr;
+}
+DriveToNote *DriveToNote::m_instance = nullptr;
+DriveToNote *DriveToNote::getInstance()
+{
+    if (DriveToNote::m_instance == nullptr)
+    {
+        DriveToNote::m_instance = new DriveToNote();
+    }
+    return DriveToNote::m_instance;
+}
+
+units::angle::degree_t DriveToNote::getNoteDirection()
+{
+    auto visiondata = DragonVision::GetDragonVision();
```

Comment:
Utilize tuple<DragonDriveTargetFinder::TARGET_INFO, Pose2d> DragonDriveTargetFinder::GetPose(DragonVision::VISION_ELEMENT item)
 to get the location of the note  then you can just create a path from the current pose (m_chassis->GetPose()) to this pose.